include/multipass/exceptions/base_qexception.h

```diff
@@ -25,13 +25,14 @@
 namespace multipass
 {
 // CRTP template base class that is the boilerplate code of QException derived classes
+// The usage of it should be CRTP derive class, that is inheriting the base calass and label itself final.
```

Comment:
```suggestion
// Derived classes should either be final or be themselves templated with the ultimate leaf type.
```

Reply:
@ricab , indeed, we discussed two ways of using this base class. Later I thought about it again, since we are suggesting users about how to use CRTP base class here, the templated derived class is no longer a CRTP and it does not make sense here. That is why I ditched that (only keep the final derived class possibility). Tell me what you think about it.

Reply:
IMO we shouldn't opt out of something just because it goes a little beyond the strict textbook concept that the term CRTP describes. Patterns often need to be adapted when employed. You could call it "generalized CRTP" if you want. 

Exceptions very often require multiple levels of inheritance. This comment precludes that, even though it would work just fine. That may cause unnecessary hurdles to future devs, instead of giving them the solution right away.

That said, this is a small matter and I am not going to block on it (but I would probably just do it myself in a PR if I came across it :wink: ). I will request that you at least fix the typos in the message, please.